### PR TITLE
Fix potential resource leak issues in LocalPageStore

### PR DESCRIPTION
**Pull request by bradyoo**
Close Files.walk stream resource in LocalPageStore

### CODE REVIEW

**Comment by bradyoo** [comment]
I also saw FileInputStream fis = new FileInputStream(p.toFile()); in get(PageId pageId, int pageOffset) but returns a channel but as far as I can see, closing the channel doesn't close the initial FileInputStream.

**Comment by bradyoo** [comment]
Actually, let me fix that as well.

File: core/client/fs/src/main/java/alluxio/client/file/cache/store/DelegatingFileChannel.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+import java.nio.MappedByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.channels.FileLock;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.channels.WritableByteChannel;
+
+/**
+ * A delegating {@link FileChannel} that also closes the FileInputStream as well.
+ */
+public class DelegatingFileChannel extends FileChannel {
```
Comment (calvinjia) [inline core/client/fs/src/main/java/alluxio/client/file/cache/store/DelegatingFileChannel.java]: Make a vanilla delegating file channel and then a subclass that does the closing?

File: core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java [hunk 2]
```diff
@@ -69,9 +69,12 @@ public LocalPageStore(LocalPageStoreOptions options) throws IOException {
...
         String.format("%s/%d/(\\d+)/(\\d+)", Pattern.quote(rootDir.toString()), mPageSize));
     try {
-      boolean invalidPage = Files.exists(rootDir) && Files.walk(rootDir)
-          .filter(Files::isRegularFile)
-          .anyMatch(path -> {
+
+      boolean invalidPage = false;
+
+      if (Files.exists(rootDir)) {
+        try (Stream<Path> pathStream = Files.walk(rootDir).filter(Files::isRegularFile)) {
```
Comment (calvinjia) [inline core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java]: Do you mind doing the filter outside of the try? (See `#getPages`)

**Comment by bradyoo** [comment]
Actually, https://stackoverflow.com/questions/8066130/should-i-close-the-filechannel seems to suggest that closing file channel does close the resource underlying file. So I'll remove the delegating bit.

**Comment by bradyoo** [comment]
Feedbacks addressed.

**Comment by calvinjia** [comment]
@bradyoo Do you plan to address closing the fileinputstream in a separate PR?

**Comment by calvinjia** [comment]
jenkins, test this please.

**Comment by bradyoo** [comment]
Failed tests: 
alluxio.server.ft.journal.TableMasterJournalIntegrationTest.journalTransformDb(alluxio.server.ft.journal.TableMasterJournalIntegrationTest)
  Run 1: TableMasterJournalIntegrationTest.journalTransformDb:198 expected:<COMPLETED> but was:<FAILED>
  Run 2: TableMasterJournalIntegrationTest.journalTransformDb:198 expected:<COMPLETED> but was:<FAILED>

... This is definitely flaky...

**Comment by bradyoo** [comment]
alluxio-bot, test this please.

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Comment by calvinjia** [comment]
alluxio-bot, cherry-pick this to branch-2.2 please.

**Pull request merged**